tests/internals/cache_metrics/cache_metrics_test.go

```diff
@@ -0,0 +1,224 @@
+	// should scale in to 8/2 = 4 replicas, irrespective of current replicas
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 4, 60, 1),
+		"replica count should be 4 after 1 minute")
+
+	// Changing Metric Value to 4, but because we have a long polling interval, the replicas number should remain the same
+	data.MonitoredDeploymentReplicas = 4
+	KubectlApplyWithTemplate(t, data, "monitoredDeploymentTemplate", monitoredDeploymentTemplate)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredDeploymentName, testNamespace, 4, 60, 1),
+		fmt.Sprintf("replica count should be %d after 1 minute", defaultMonitoredDeploymentReplicas))
+
+	t.Log("waiting 30s to allow propagation of changes")
+
+	// Let's sleep at least 30s
+	// the usual setting for `horizontal-pod-autoscaler-sync-period` is 15s)
+	time.Sleep(30 * time.Second)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 4, 60, 1),
+		"replica count should be 4 after 1 minute")
+
+	KubectlDeleteWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+}
+
+func testDirectQuery(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing query metrics directly ---")
+
+	KubectlApplyWithTemplate(t, data, "deploymentTemplate", deploymentTemplate)
+	data.MonitoredDeploymentReplicas = defaultMonitoredDeploymentReplicas
+	KubectlApplyWithTemplate(t, data, "monitoredDeploymentTemplate", monitoredDeploymentTemplate)
+
+	// initial replica count for a deployment
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 1, 60, 1),
+		"replica count should be 1 after 1 minute")
+
+	// initial replica count for a monitored deployment
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredDeploymentName, testNamespace, defaultMonitoredDeploymentReplicas, 60, 1),
+		fmt.Sprintf("replica count should be %d after 1 minute", defaultMonitoredDeploymentReplicas))
+
+	data.EnableUseCachedMetrics = false
+	KubectlApplyWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+
+	// Metric Value = 8, DesiredAverageMetricValue = 2
+	// should scale in to 8/2 = 4 replicas, irrespective of current replicas
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 4, 60, 1),
+		"replica count should be 4 after 1 minute")
+
+	// Changing Metric Value to 4, deployment should scale to 2
+	data.MonitoredDeploymentReplicas = 4
+	KubectlApplyWithTemplate(t, data, "monitoredDeploymentTemplate", monitoredDeploymentTemplate)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredDeploymentName, testNamespace, 4, 60, 1),
+		fmt.Sprintf("replica count should be %d after 1 minute", defaultMonitoredDeploymentReplicas))
+
+	t.Log("waiting 30s to allow propagation of changes")
+
+	// Let's sleep at least 30s
+	// the usual setting for `horizontal-pod-autoscaler-sync-period` is 15s)
+	time.Sleep(30 * time.Second)
```

Comment:
do we need this here? the `WaitForDeploymentReplicaReadyCount` already waits some time. We could just extend the timeout from 1 minute to 2. WDYT?